stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/message/attachment/UploadAttachmentsWorker.kt

```diff
@@ -1,19 +1,40 @@
 package io.getstream.chat.android.offline.message.attachment
 
 import android.content.Context
+import io.getstream.chat.android.client.ChatClient
 import io.getstream.chat.android.client.errors.ChatError
 import io.getstream.chat.android.client.models.Attachment
 import io.getstream.chat.android.client.utils.Result
-import io.getstream.chat.android.offline.ChatDomain
 import io.getstream.chat.android.offline.ChatDomainImpl
 
 internal class UploadAttachmentsWorker(private val appContext: Context) {
-    suspend fun uploadAttachmentsForMessage(channelType: String, channelId: String, messageId: String): Result<Unit> {
+    suspend fun uploadAttachmentsForMessage(
+        channelType: String,
+        channelId: String,
+        messageId: String,
+        chatDomain: ChatDomainImpl,
+        chatClient: ChatClient
+    ): Result<Unit> {
         return try {
-            val domainImpl = (ChatDomain.instance() as ChatDomainImpl)
+            val domainImpl = chatDomain.apply {
+                if (!isRepositoryInitialized) {
+                    chatClient.setUserWithoutConnectingIfNeeded()
+
+                    chatClient.getCurrentUser()?.let(::initRepositoryFacade)
```

Comment:
We already have an approach on how to init data with a user, and how to init all required stuff (repo with cache in this case), so we shouldn't do anything else except setting a user here.

Reply:
We just need to somehow wait until ChatClient notifies ChatDomain about new user

Reply:
I see it's done in synchronous way. Should work fine, but better to debug
```kotlin
private fun notifySetUser(user: User) {
        preSetUserListeners.forEach { it(user) }
    }
```

Reply:
Ok. Now this flag is removed and no manual initialization of ChatDomain is happnening